pkg/resto/registry.go

```diff
@@ -0,0 +1,157 @@
+package resto
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/registry/client/auth"
+	"github.com/docker/distribution/registry/client/transport"
+	"github.com/docker/docker/api/types"
+	dd "github.com/docker/docker/distribution"
+	"github.com/docker/docker/registry"
+	digest "github.com/opencontainers/go-digest"
+)
+
+type myreference string
+
+func (m myreference) String() string {
+	return string(m)
+}
+
+func (m myreference) Name() string {
+	return string(m)
+}
+
+// MediaTypeConfig is the media type used for configuration files.
+const MediaTypeConfig = "application/vndr.docker.config"
```

Comment:
Looking at some existing media type, this one should be something like `application/vnd.docker.app.config.tar`

Reply:
It's not app-specific, the aim is to target arbitrary config files with resto, and it's not a tarball.

Reply:
then `application/vnd.docker.config.tar` (my point was *mainly* about the `tar` part)

Reply:
It's not a tar!